fix(deploy): prod migrate-prod fails on @/lib/db alias in signup.ts

## Problem

Deploy run [24636786335](https://github.com/ingamartinez/personal-financial-dashboard/actions/runs/24636786335) failed at the migrate step with:

```
error: Cannot find module '@/lib/db' from '/srv/findash/app/releases/.../src/lib/auth/signup.ts'
Bun v1.3.12 (Linux x64)
```

## Root cause

PR #267 copies `src/lib/auth/signup.ts` to the release stage so `scripts/migrate-prod.ts` can import `copyCategorySeedsToUser` + `copyRuleSeedsToUser`. But `signup.ts` uses `@/lib/db` and `@/lib/db/schema` — TypeScript path aliases resolved by Next's build, not by bun at runtime. On the droplet there's no tsconfig to map `@/*` → `src/*`, so bun can't find the module.

## Fix

Replace the two `@/` imports in `signup.ts` with relative paths (`../db`, `../db/schema`). Since both dev and prod preserve the `src/lib/auth/signup.ts` layout, relative imports resolve identically in both contexts.

## Blast radius

**Prod is currently stuck** — the last successful deploy was for PR #265 (commit `0db31db`). PR #267 (commit `e4e61b38`) is on main but the droplet is still running the #265 build. No user-visible regression — the existing per-user-telemetry + /admin features stayed live because the symlink flip only happens AFTER migrate succeeds.

--- a/src/lib/auth/signup.ts
+++ b/src/lib/auth/signup.ts
@@ -1,11 +1,15 @@
 import { sql } from "drizzle-orm";
-import { db as defaultDb, type DB } from "@/lib/db";
+// Relative imports (not `@/lib/...` aliases) are required because this file
+// is copied verbatim into the prod release by deploy.yml and executed there
+// by bun, which has no tsconfig to resolve path aliases. Keeping these
+// relative makes the same file work in dev and on the droplet.
+import { db as defaultDb, type DB } from "../db";
 import {
   categories,
   categorySeeds,
   classificationRuleSeeds,
   classificationRules,
-} from "@/lib/db/schema";
+} from "../db/schema";
 
 /**
  * Copy every row from `category_seeds` into `categories` with the new user's

